provider/pkg/provider/provider.go

```diff
@@ -376,6 +376,15 @@ func (k *kubeProvider) Configure(_ context.Context, req *pulumirpc.ConfigureRequ
 
 	vars := req.GetVariables()
 
+	//
+	// Check that kubeconfig is present.
+	//
+
+	if k.config == nil {
+		return nil, fmt.Errorf("unable to locate kubeconfig. Make sure you have: \n\n" +
+			" \t • Setup the provider as per https://pulumi.io/install/kubernetes.html \n\n")
```

Comment:
```suggestion
			" \t • set up the provider as per https://pulumi.io/install/kubernetes.html \n\n")
```

Reply:
Is k.config checking for the environment variable as well?

Reply:
Oh wow, good catch @stack72. Coffee hasn't kicked in yet, and I just realized that this will always fail. This check needs to happen later in the Configure method.

Reply:
Yes, good catch, thank you, I believe that the first commit here is closer to the right location for this check to happen, working on a check there that will give this message if $KUBECONFIG isn't set, and pass along a more helpful error if the cluster is unreachable for another reason.